PokeAlarm/Events/EggEvent.py

```diff
@@ -44,6 +44,10 @@ def __init__(self, data):
             str, data.get('description'), Unknown.REGULAR).strip()
         self.gym_image = check_for_none(
             str, data.get('url'), Unknown.REGULAR)
+        self.gym_sponsor = check_for_none(
+            int, data.get('sponsor'), Unknown.SMALL)
```

Comment:
what is the `sponsor` field, and how is it determined?

Reply:
Sponsor is the sponsor flag for a gym, as in the companies sponsoring gyms. We do not have any in UK so cannot verify how this works - this was originally a PR to my fork, that I guessed would be useful on stock.

Reply:
what are the possible values of the sponsor field?  Is this in Proto?

Reply:
Yes, looks like 23 is current max value used

Reply:
@evilmoses Is it a value derived from the pgo api or is it determined by the scanner?

Reply:
Pogo protos

Reply:
If we are going to this 'bool' - Python considers '0' to be 'falsey', so that should give us False for 0 and True for everything else. ('None' will still be treated as unknown).

In the future someone may want to expand upon this to filter by specific sponsor, but I'm fine with this for now.

Reply:
Um, that's as intended. 0 is false, >0 is true and None is unknown. For non sponsored gyms the value is always 0. In the events though, we will always return into, not bool. Book is for the filter only.